cEP-0024.md

```diff
@@ -0,0 +1,107 @@
+# Implement GitMate automations and plugins for coala
+
+| Metadata |                                                       |
+| -------- | ----------------------------------------------------- |
+| cEP      | 24                                                    |
+| Version  | 1.0                                                   |
+| Title    | Implement Gitmate automations and plugins for coala   |
+| Authors  | Vamshi Krishna Bommerla <mailto:bommvams72@gmail.com> |
+| Status   | Proposed                                              |
+| Type     | Process                                               |
+
+## Abstract
+
+This cEP describes the improvements to the coala workflow and the necessary
+automation to realise it, as a part of the
+[GSoC project](https://summerofcode.withgoogle.com/projects/#5344316227059712).
+
+## Introduction
+
+[GitMate](https://gitmate.io/) is an automation tool for developers. While it
+works well, there are a number of missing features which would save time of
+coala community, for example automatically close PRs from master of forks. Many
+but coala needs more plugins which support coala development workflow. This
+project adds plugins to auto-review or auto-reject PRs, assign issues. A team
+class, representing GitHub and GitLab teams, will be implemented in IGitt.
+
+## New Features
+
+### Auto-assign issue when it is mentioned in any of PR commits
+
+When a PR is opened in the repo and if any of its commit message
+mentions (Fixes, Closes) any unassigned issues and there is no CI failure, auto
+assign commit author to the mentioned unassigned issues. Comment on the
+PR (configurable) if someone else is working on the mentioned issue.
+
+### Auto-reject PR opened on master of forks
+
+Normally, contributors should make changes in different branch from master
+branch of the forked repo, and submit a PR. But, sometimes coala receives PRs
+from master branch of the forked repo. To solve this problem, coala needs
+mechanism by which GitMate automatically closes those PRs and comments on the
+PR informing the PR author to work on a different branch. Comment message will
+be configurable.
+
+### Auto-mark PR with necessary labels as per the reviews done
+
+When a reviewer reviews a PR, he/she either approves changes or request changes
+or comments something and leaves it there, but the PR would be still in the
+review queue. This will be a part of ack plugin. A setting to limit this plugin
+to certain teams will be added. GitMate will automatically mark labels as per
+the following conditions:
+
+- WIP label conditions:
+  1. Reviewer requests changes
+- Approve label conditions:
+  1. Reviewer approves changes
+  2. Anyone comments on the PR which has a CI failure
```

Comment:
If a PR has CI failure then wouldn't it be in WIP (irrespective of whether anyone comments on the PR or not)?

Could you explain a bit.